### DISCUSSION TITLE
Report - (3rd Sem.) Week 3 📢

### DISCUSSION THREAD

**Discussion by Nahlam4** [SE-TINF22B6/DHBWhub#94] [Weekly Report]
Welcome to the third report of our progress. We are excited to share the latest news with you! 🥳

### Documentation and Planning 📝 
This week, we further planned and sketched more processes of use cases, user activities and sequence diagrams to understand the underlining of our project even better. You can find our software requirement specifications [here](https://github.com/SE-TINF22B6/DHBWhub/blob/master/docs/SOFTWARE_REQUIREMENTS_SPECIFICATION.md)! Let's take a look at some important diagrams:  

#### Use case diagram

✅ We have revised our general use case diagram of the last week and separated it into several sections for easier understanding. The upgraded diagram looks like this:

![UsecaseDiagramGeneral](https://github.com/SE-TINF22B6/DHBWhub/blob/master/docs/diagrams/UsecaseDiagramGeneral.drawio.png?raw=true)

#### User activity diagrams

✅ This UML activity diagram depicts the user opening a post in the thread-view. The user starts on the homepage and the 
following course of actions depends on whether he is logged in or not. The login process will be explained in
the graph after the following:

![uml-er](https://github.com/SE-TINF22B6/DHBWhub/blob/master/docs/diagrams/UserActivityThreadView.drawio.png?raw=true)

✅ We visualized the login process (including validations) as a sub-sequence offering the choice to sign in via 3rd-Party or register a
new account although latter is depicted in more detail on the third graph:

![uml-userActivityLogin](https://github.com/SE-TINF22B6/DHBWhub/blob/master/docs/diagrams/UserActivityLogin.drawio.png?raw=true)

✅ Here is a rough presentation of our registration process. For the last two activities, we have simplified the cancel process through closing windows or similar processes.:

![uml-er](https://github.com/SE-TINF22B6/DHBWhub/blob/master/docs/diagrams/UserActivityRegister.drawio.png?raw=true)

#### Sequence diagrams
In order to display the technical relations of our website, the backend and other third parties, we created sequence diagrams
which could help us implementing these functions by depicting the whole process behind them. 

✅ This sequence diagram shows the process of signing onto the website through 3rd-party-APIs like Google's Sign-in:

![seq-diagram](https://github.com/SE-TINF22B6/DHBWhub/blob/master/docs/diagrams/LoginSequenceDiagram.drawio.png?raw=true)

✅ After signing in, the creation of a post by a user could look like this:

![seq-diagram](https://github.com/SE-TINF22B6/DHBWhub/blob/master/docs/diagrams/CreatePostSequenceDiagram.drawio.png?raw=true)

### Further Design 💻

✅ We continued creating mockups and are proud to present our Thread/Post page:

![image](https://github.com/SE-TINF22B6/DHBWhub/assets/118728701/e9b2612e-d5e3-4282-8345-99d8278c349e)

### Sprint 2: What have we achieved ❓  What's to come❔
- ✅Agreeing on our login-options and the way we persist account data
- ✅Using Figma for further design of the post-page
- ✅Plan Sign-in process
- ✅Implementation of ER-diagram in the database
- ✅Setup Server for hosting the database
    - ✅Setup database on the server
- 🚧 Building the web interface
    - 🚧Creating Signup/Login-window
    - ✅Creating navbar 
- 🚧Discussing possible API's 
- 🚧Removing Herobrine 

Until next week 💯

Your DHBWhub Team

**Comment by dolerich-hirnfiedler** [comment]
Your Blog Post is very informative.
I am also curious when we will receive the results of the Digitaltechnik exam from Prof. Röthig. 
Your diagrams seem very nice, and I am interested in the tools you used for creating them. 
On the first diagram, it seems like after visiting the website, there is some kind of action for signup/login. I find it confusing to call this B. And later on explain B. 
You could do the same for visiting a website. Call it B1 and then draw an activity-diagram for visiting a website.
![browse-website-example](https://github.com/SE-TINF22B6/DHBWhub/assets/87950162/d3faf235-d041-4903-abe4-fc55a61ee67d)

I think it would be clearer if you called the action sign in and register, and later on used the diagram in B to explain how this process works in detail.

**Reply by giomskii** [reply]
Thanks for the feedback, we have updated and specified the description to clear up some misconception. Moreover, we use draw.io for the diagrams 😄

**Comment by Challenger3010** [comment]
very detailed description of your progress i find your design very appealing and am curious about the result. 😄

**Comment by fehlend0** [comment]
Dear DHBWhub Team,

It's great to see your project moving forward. 

Your focus on further planning and sketching processes, including use cases, user activities, and sequence diagrams, is very positive for understanding and developing your project. We also admired your user activity diagrams and their visual representation of the login and registration processes. We found them well-structured and self-explaining. As for your mockups, the Thread/Post page mockup looks promising and provides a good insight into the user interface's direction. It's also great to see that you've achieved several key milestones. 

We look forward to hearing more about your project next week!

Your Statify Team
